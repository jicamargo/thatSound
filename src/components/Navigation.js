import React from 'react';
import { NavLink } from 'react-router-dom';
import logoIcon from '../assets/wave-sound.png';

function Navigation() {
  return (
    <nav>
      <div className="div-logo">
        <img className="logo-img" src={logoIcon} alt="App Icon" />
        <span className="logo">thatSound</span>
      </div>
      <ul className="navbar">
        <li>
          <NavLink to="/Sounds" className="menu-item" activeclassname="active">Sounds</NavLink>
        </li>
        <li>
          <NavLink to="/Details" className="menu-item" activeclassname="active">Details</NavLink>
        </li>
        <li>
          <NavLink to="/About" className="menu-item" activeclassname="active">About</NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default Navigation;
