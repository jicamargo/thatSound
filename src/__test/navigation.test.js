import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';

describe('Navigation', () => {
  it('should render Navigation component correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
