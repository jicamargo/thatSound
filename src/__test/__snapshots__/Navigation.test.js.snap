// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render Navigation component correctly 1`] = `
<div>
  <nav>
    <div
      class="div-logo"
    >
      <img
        alt="App Icon"
        class="logo-img"
        src="wave-sound.png"
      />
      <span
        class="logo-text"
      >
        <span
          class="white-text"
        >
          that
        </span>
        <span
          class="blue-text"
        >
          Sound
        </span>
      </span>
    </div>
    <ul
      class="navbar"
    >
      <li>
        <a
          activeclassname="active"
          class="menu-item"
          href="/Sounds"
        >
          Sounds
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          class="menu-item"
          href="/Details"
        >
          Details
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          class="menu-item"
          href="/About"
        >
          About
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
